Plugins do not yet support ruby 2.4 and therefore breaks on sensu 0.29

After upgrading sensu to 0.29 this plugin doesn't build properly.

The error I see from the check is this:

```
/opt/sensu/embedded/lib/ruby/site_ruby/2.4.0/rubygems.rb:270:in `find_spec_for_exe': can't find gem sensu-plugins-percona (>= 0.a) (Gem::GemNotFoundException) from /opt/sensu/embedded/lib/ruby/site_ruby/2.4.0/rubygems.rb:298:in `activate_bin_path' from /opt/sensu/embedded/bin/check-wsrep-ready.rb:22:in `<main>'
```

If I try to install the plugin this is what I get:

```
[SENSU-INSTALL] installing Sensu plugins ...
[SENSU-INSTALL] determining if Sensu gem 'sensu-plugins-percona' is already installed ...
false
[SENSU-INSTALL] Sensu plugin gems to be installed: ["sensu-plugins-percona"]
[SENSU-INSTALL] installing Sensu gem 'sensu-plugins-percona'
Fetching: mysql-2.9.1.gem (100%)
Building native extensions.  This could take a while...
ERROR:  Error installing sensu-plugins-percona:
	ERROR: Failed to build gem native extension.

    current directory: /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/mysql-2.9.1/ext/mysql_api
/opt/sensu/embedded/bin/ruby -r ./siteconf20170412-9919-1iffwnx.rb extconf.rb
checking for mysql_ssl_set()... yes
checking for rb_str_set_len()... yes
checking for rb_thread_start_timer()... no
checking for mysql.h... yes
creating Makefile

current directory: /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/mysql-2.9.1/ext/mysql_api
make "DESTDIR=" clean

current directory: /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/mysql-2.9.1/ext/mysql_api
make "DESTDIR="
compiling mysql.c
mysql.c: In function ‘stmt_bind_result’:
mysql.c:1320:74: error: ‘rb_cFixnum’ undeclared (first use in this function)
  else if (argv[i] == rb_cNumeric || argv[i] == rb_cInteger || argv[i] == rb_cFixnum)
                                                                          ^
mysql.c:1320:74: note: each undeclared identifier is reported only once for each function it appears in
Makefile:241: recipe for target 'mysql.o' failed
make: *** [mysql.o] Error 1

make failed, exit code 2

Gem files will remain installed in /opt/sensu/embedded/lib/ruby/gems/2.4.0/gems/mysql-2.9.1 for inspection.
Results logged to /opt/sensu/embedded/lib/ruby/gems/2.4.0/extensions/x86_64-linux/2.4.0/mysql-2.9.1/gem_make.out
[SENSU-INSTALL] failed to install Sensu gem 'sensu-plugins-percona'
[SENSU-INSTALL] you can run the sensu-install command again with --verbose for more info
[SENSU-INSTALL] please take note of any failure messages above
[SENSU-INSTALL] make sure you have build tools installed (e.g. gcc)
[SENSU-INSTALL] trying to determine the Sensu plugin homepage for sensu-plugins-percona ...
homepage: https://github.com/sensu-plugins/sensu-plugins-percona
```

And looking online this seems to be the cause:

**The mysql gem is one of many gems which doesn't like the unification of the integer types in Ruby 2.4.**

Reference: http://stackoverflow.com/questions/41520996/mysql-gem-install-error-on-ruby-2-4

P.S. I do have libmysqlclient-dev installed.

--- a/.github/PULL_REQUEST_TEMPLATE.md
+++ b/.github/PULL_REQUEST_TEMPLATE.md
@@ -12,7 +12,7 @@
 
 - [ ] RuboCop passes
 
-- [ ] Existing tests pass 
+- [ ] Existing tests pass
 
 #### New Plugins
 
@@ -24,5 +24,4 @@
 
 #### Purpose
 
-#### Known Compatablity Issues
-
+#### Known Compatibility Issues

--- a/.travis.yml
+++ b/.travis.yml
@@ -8,6 +8,7 @@ rvm:
 - 2.1
 - 2.2
 - 2.3.0
+- 2.4.1
 notifications:
   email:
     recipients:

--- a/.travis.yml
+++ b/.travis.yml
@@ -31,4 +31,5 @@ deploy:
     rvm: 2.1
     rvm: 2.2
     rvm: 2.3.0
+    rvm: 2.4.1
     repo: sensu-plugins/sensu-plugins-percona

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,9 +1,17 @@
-#Change Log
+# Change Log
 This project adheres to [Semantic Versioning](http://semver.org/).
 
 This CHANGELOG follows the format listed at [Keep A Changelog](http://keepachangelog.com/)
 
 ## [Unreleased]
+
+### Removed
+- removed `mysql` gem dep as we have `mysql2` and `mysql` does not work on ruby 2.4 and has been abandoned. (@majormoses)
+
+### Changed
+- check-percona-cluster-size.rb: converted to use `mysql2` gem (@majormoses)
+- check-wsrep-ready.rb: converted to use `mysql2` gem (@majormoses)
+
 ### Added
 - ruby 2.4 testing to travis
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,8 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 This CHANGELOG follows the format listed at [Keep A Changelog](http://keepachangelog.com/)
 
 ## [Unreleased]
+### Added
+- ruby 2.4 testing to travis
 
 ## [1.0.0] - 2016-09-25
 ### Added

--- a/bin/check-percona-cluster-size.rb
+++ b/bin/check-percona-cluster-size.rb
@@ -9,7 +9,7 @@
 # Based on the MySQL Health Plugin by Panagiotis Papadomitsos
 #
 # Depends on mysql:
-# gem install mysql
+# gem install mysql2
 #
 # Released under the same terms as Sensu (the MIT license); see LICENSE
 # for details.

--- a/bin/check-percona-cluster-size.rb
+++ b/bin/check-percona-cluster-size.rb
@@ -15,7 +15,7 @@
 # for details.
 
 require 'sensu-plugin/check/cli'
-require 'mysql'
+require 'mysql2'
 require 'inifile'
 
 class CheckPerconaClusterSize < Sensu::Plugin::Check::CLI
@@ -51,15 +51,16 @@ class CheckPerconaClusterSize < Sensu::Plugin::Check::CLI
     if config[:ini]
       update_config
     end
-    db = Mysql.real_connect(config[:hostname], config[:user], config[:password], config[:database])
-    cluster_size = db
-                   .query("SHOW GLOBAL STATUS LIKE 'wsrep_cluster_size'")
-                   .fetch_hash
-                   .fetch('Value')
-                   .to_i
+    db = Mysql2::Client.new(
+      host:     config[:hostname],
+      username: config[:user],
+      password: config[:password],
+      database: config[:database]
+    )
+    cluster_size = db.query("SHOW GLOBAL STATUS LIKE 'wsrep_cluster_size'").first['Value'].to_i
     critical "Expected to find #{config[:expected]} nodes, found #{cluster_size}" if cluster_size != config[:expected].to_i
     ok "Expected to find #{config[:expected]} nodes and found those #{cluster_size}" if cluster_size == config[:expected].to_i
-  rescue Mysql::Error => e
+  rescue Mysql2::Error => e
     critical "Percona MySQL check failed: #{e.error}"
   ensure
     db.close if db

--- a/bin/check-wsrep-ready.rb
+++ b/bin/check-wsrep-ready.rb
@@ -13,7 +13,7 @@
 #
 # DEPENDENCIES:
 #   gem: sensu-plugin
-#   gem: mysql
+#   gem: mysql2
 #
 # USAGE:
 #

--- a/bin/check-wsrep-ready.rb
+++ b/bin/check-wsrep-ready.rb
@@ -28,7 +28,7 @@
 #
 
 require 'sensu-plugin/check/cli'
-require 'mysql'
+require 'mysql2'
 
 class CheckWsrepReady < Sensu::Plugin::Check::CLI
   option :user,
@@ -49,11 +49,16 @@ class CheckWsrepReady < Sensu::Plugin::Check::CLI
          default: 'localhost'
 
   def run
-    db = Mysql.real_connect(config[:hostname], config[:user], config[:password], config[:database])
-    wsrep_ready = db.query("SHOW STATUS LIKE 'wsrep_ready';").fetch_hash.fetch('Value')
+    db = Mysql2::Client.new(
+      host:     config[:hostname],
+      username: config[:user],
+      password: config[:password],
+      database: config[:database]
+    )
+    wsrep_ready = db.query("SHOW STATUS LIKE 'wsrep_ready';").first['Value']
     critical "WSREP Ready is not ON. Is #{wsrep_ready}" if wsrep_ready != 'ON'
     ok 'Cluster is OK!' if wsrep_ready == 'ON'
-  rescue Mysql::Error => e
+  rescue Mysql2::Error => e
     critical "Percona MySQL check failed: #{e.error}"
   ensure
     db.close if db

--- a/sensu-plugins-percona.gemspec
+++ b/sensu-plugins-percona.gemspec
@@ -30,8 +30,7 @@ Gem::Specification.new do |s|
   s.version                = SensuPluginsPercona::Version::VER_STRING
 
   s.add_runtime_dependency 'sensu-plugin', '~> 1.2'
-  s.add_runtime_dependency 'mysql',        '2.9.1'
-  s.add_runtime_dependency 'mysql2',       '0.4.6'
+  s.add_runtime_dependency 'mysql2',       '0.4.8'
   s.add_runtime_dependency 'inifile',      '3.0.0'
 
   s.add_development_dependency 'bundler',                   '~> 1.7'

--- a/sensu-plugins-percona.gemspec
+++ b/sensu-plugins-percona.gemspec
@@ -31,7 +31,7 @@ Gem::Specification.new do |s|
 
   s.add_runtime_dependency 'sensu-plugin', '~> 1.2'
   s.add_runtime_dependency 'mysql',        '2.9.1'
-  s.add_runtime_dependency 'mysql2',       '0.3.18'
+  s.add_runtime_dependency 'mysql2',       '0.4.6'
   s.add_runtime_dependency 'inifile',      '3.0.0'
 
   s.add_development_dependency 'bundler',                   '~> 1.7'

